pgmpy/factors/continuous/ContinuousFactor.py

```diff
@@ -0,0 +1,259 @@
+        ...     return (np.power(x, 1)*np.power(y, 2))/beta(x, y)
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> drichlet_factor = ContinuousFactor(['x', 'y'], drichlet_pdf)
+        >>> drichlet_factor.variables
+        ['x', 'y']
+        >>> copy_factor = drichlet_factor.copy()
+        copy_factor.variables
+
+        """
+        return ContinuousFactor(self.scope(), self.pdf)
+
+    def reduce(self, values, inplace=True):
+        """
+        Reduces the factor to the context of the given variable values.
+
+        Parameters
+        ----------
+        values: list, array-like
+            A list of tuples of the form (variable_name, variable_value).
+
+        inplace: boolean
+            If inplace=True it will modify the factor itself, else would return
+            a new ContinuosFactor object.
+
+        Returns
+        -------
+        ContinuousFactor or None: if inplace=True (default) returns None
+                                  if inplace=False returns a new `Factor` instance.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from scipy.special import beta
+        >>> def custom_pdf(x, y, z):
+        ...     return z*(np.power(x, 1)*np.power(y, 2))/beta(x, y)
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> custom_factor = ContinuousFactor(['x', 'y', 'z'], custom_pdf)
+        >>> custom_factor.variables
+        ['x', 'y', 'z']
+        >>> custom_factor.assignment(1, 2, 3)
+        24.0
+
+        >>> custom_factor.reduce([('y', 2)])
+        >>> custom_factor.variables
+        ['x', 'z']
+        >>> custom_factor.assignment(1, 3)
+        24.0
+        """
+        if not isinstance(values, list):
+            raise TypeError("values: Expected type list or array-like,\
+                             got type {var_type}".format(var_type=type(values)))
+
+        phi = self if inplace else self.copy()
+
+        var_to_remove = [var for var, value in values]
+        var_to_keep = [var for var in self.variables if var not in var_to_remove]
+
+        reduced_var_index = [(self.variables.index(var), value) for var, value in values]
+        pdf = self.pdf
+
+        def reduced_pdf(*args, **kwargs):
+            reduced_args = list(args)
+            reduced_kwargs = kwargs.copy()
+
+            if reduced_args:
+                for index, value in reduced_var_index:
+                    reduced_args.insert(index, value)
+            if reduced_kwargs:
+                for var, value in values:
+                    reduced_kwargs[var] = value
+            if reduced_args and reduced_kwargs:
+                reduced_args = [arg for arg in reduced_args if arg not in reduced_kwargs.values()]
+
+            return pdf(*reduced_args, **reduced_kwargs)
+
+        phi.variables = var_to_keep
+        phi.pdf = reduced_pdf
+
+        if not inplace:
+            return phi
+
+    def marginalize(self, variables, inplace=True):
```

Comment:
@ankurankan For ```product``` and ```divide``` simply, I think simply dividing and multiplying the pdfs would not ensure that the new pdf always integrated to 1 for all the variables. Don't we need proper definitions for it so that all operations ensure that new pdf is valid?